Feature/gfsv16intocoupled

jkuang-exploro opened the pull request:
Script update.

Additional code and logics blocks from gfsv16 are included in the modularized scripts.

Next step is to add them into the namelists

lgannoaa committed:
add gfs gempak downstream jobs into def file

KateFriedman-NOAA committed:
Update develop branch to latest version of GFSv16 implementation branch

lgannoaa committed:
Merge branch 'feature/gfsv16b' of https://github.com/lgannoaa/global-workflow into feature/gfsv16b

Commit:
Issue #94 add waves-prdgen, ICE->ICEC, Sleep in gempak script

Commit:
Issue #94 solving a conflict

RussTreadon-NOAA committed:
Issue #1:  update name of DA jobs and scripts in accordance with WCOSS Implementation Standards

KateFriedman-NOAA committed:
Merge pull request #128 from RobertoPadilla-NOAA/feature/gfsv16_wave_prdgen

Feature/gfsv16 wave prdgen

KateFriedman-NOAA committed:
Merge branch 'feature/gfsv16b' into feature/gfsv16b-wave

KateFriedman-NOAA committed:
Increase walltime for new wavepost jobs

KateFriedman-NOAA committed:
Merge remote-tracking branch 'upstream/feature/gfsv16b-wave' into feature/gfsv16b-splitwavepost

JessicaMeixner-NOAA committed:
Merge pull request #4 from KateFriedman-NOAA/feature/gfsv16b-splitwavepost

Sync with feature/gfsv16b and small updates to split wave post changes

KateFriedman-NOAA committed:
Merge pull request #124 from JessicaMeixner-NOAA/feature/gfsv16b-wave

Splitting post jobs for waves

RussTreadon-NOAA committed:
Issue #1:  correct DA typos in sorc/link_fv3gfs.sh

yangfanglin committed:
deleted  relocate_mv_nvortex.fd since storm relocation is no longer needed.
modified build_tropcy_NEMS.sh to remove references to relocate_mv_nvortex

lgannoaa committed:
Restructured ecflow - up to post step

JessicaMeixner-NOAA committed:
adding the line to go back a day for RTOFS for the if not NCO
section because RTOFS will not be available until 06 cycle

KateFriedman-NOAA committed:
Merge pull request #130 from JessicaMeixner-NOAA/bugfix/rtofs

Bugfix for rtofs

RussTreadon-NOAA committed:
Issue #1:  add fhrgrp and fhrlst back to gfsawips in setup_workflow.py (bugfix)

JessicaMeixner-NOAA committed:
updates for by hour post

lgannoaa committed:
Adding wafs wave and downstream jobs

GuangPingLou-NOAA committed:
Issue #131 added a path to DBNROOT

GuangPingLou-NOAA committed:
Issue #131 reduce scripts output to logfile

lgannoaa committed:
Merge remote-tracking branch 'upstream/feature/gfsv16b' into feature/gfsv16b

KateFriedman-NOAA committed:
Small updates:
- new UPP tag
- new GLDAS tag
- new WAFS tag
- new module for WAFS
- EE2 updates to awips scripts
- added WAFS to archival
- break downstream and WAFS archival into separate gfs_downstream tarball
- update gfsarch dependencies to wait for all wavepost jobs to complete

KateFriedman-NOAA committed:
Update WAFS tag to gfs_wafs.v6.0.6

lgannoaa committed:
ecflow full day cycle included

KateFriedman-NOAA committed:
Fix for running prep on Hera

KateFriedman-NOAA committed:
Merge pull request #135 from KateFriedman-NOAA/feature/gfsv16b-down

Small updates to downstream jobs and tag updates

RussTreadon-NOAA committed:
Issue #1:  update to UPP tag "upp_gfsv16_release.v1.0.15"

GuangPingLou-NOAA committed:
Issue #131 Unify dbn_alert path

GuangPingLou-NOAA committed:
Merge remote-tracking branch 'upstream/feature/gfsv16b' into feature/gfsv16b

KateFriedman-NOAA committed:
Issue #1 - fix to link_fv3gfs.sh for new GLDAS tag

KateFriedman-NOAA committed:
Issue #1 - update to UFS_UTILS ops-gfsv16.0.0 tag

KateFriedman-NOAA committed:
Issue #1 - update SEND variables and add DBNROOT to base configs and add check to build_enkf_chgres_recenter_nc.sh for GSI build

GuangPingLou-NOAA committed:
Merge remote-tracking branch 'upstream/feature/gfsv16b' into feature/gfsv16b

JessicaMeixner-NOAA committed:
updates for optimizing point jobs

yangfanglin committed:
renamed:    scripts/exglobal_fcst_nemsfv3gfs.sh -> scripts/exglobal_forecast.sh
and modified  jobs/JGLOBAL_FORECAST parm/config/config.fcst

yangfanglin committed:
renamed:    jobs/JGFS_POSTSND -> jobs/JGFS_ATMOS_POSTSND
 	renamed:    scripts/exgfs_postsnd.sh -> scripts/exgfs_atmos_postsnd.sh
 	modified:   docs/archive/README_bufr
 	            driver/product/run_postsnd.sh
 	            driver/product/run_postsnd.sh.cray
 	            driver/product/run_postsnd.sh.dell
 	            driver/product/run_postsnd.sh.hera
 	            driver/product/run_postsnd.sh.jet
 	            parm/config/config.postsnd

yangfanglin committed:
In anticipating changes from the GLDAS repo : renaming JGDAS_GLDAS to JGDAS_ATMOS_GLDAS, and exgdas_gldas.sh to exgdas_atmos_gldas.sh

modified:   driver/gdas/test_gdas_gldas.sh jobs/rocoto/gldas.sh parm/config/config.gldas sorc/link_fv3gfs.sh

KateFriedman-NOAA committed:
Rename scripts to match ecf script naming convention. Add SENDDBN and DBNROOT. Update GLDAS tag.

KateFriedman-NOAA committed:
Update EMC_verif-global tag to verif_global_v1.11.0

BoiVuong-NOAA committed:
Updated jobs name

WenMeng-NOAA committed:
1)Update sorc/checkout.sh with new UPP tag upp_gfsv16_release.v1.0.16.
2)Update sorc/link_fv3gfs.sh with new file name convention for jjob and ex-script of post processing part.

WenMeng-NOAA committed:
Update Externals.cfg with new UPP tag upp_gfsv16_release.v1.0.16.

BoiVuong-NOAA committed:
Updated scripts

BoiVuong-NOAA committed:
Updated drivers and release notes

lgannoaa committed:
Merge remote-tracking branch 'upstream/feature/gfsv16b' into feature/gfsv16b

KateFriedman-NOAA committed:
Fixing spelling mistake in config.gldas

lgannoaa committed:
Merge feature_gfsv16b into feature_ccpp for testing

KateFriedman-NOAA committed:
Name change for tropcy scripts and update WAFS tag

KateFriedman-NOAA committed:
Script name updates for sfc_prep and tracker

KateFriedman-NOAA committed:
Issue #5 - sync merge with feature/gfsv16b_jobname

KateFriedman-NOAA committed:
Update config.base.nco.static with config.base.emc.dyn changes

lgannoaa committed:
Modify each ecflow script with old j-job name for test.

Modify the following in each definition file.
- ecen family require new extern from previous cycle
- Job jgfs_atmos_npoess_pgrb2_0p5deg trigger changed to be (requested
  from code manager):
    1. jgfs_atmos_post_manager:release_post180
- Job jgdas_enkf_select_obs trigger changed to be:
    1. previous cycle enkf post complete
    2. current cycle jgdas_atmos_analysis_calc complete (new job)

Tested as of 9/21/2020 before production switch with following condition:
1. Known issue in wafs gcip. Job failed. Waiting for code manager to
fix.
2. obsproc testing still on going.
3. Code manager still updating j-job and ex-script names. Testing in
hold until package is ready and WCOSS availability.

lgannoaa committed:
As requested from management, point ufs-weather-model to development:
b8c5c22b2a2effe7b925fae1fa449ddec96be848
git submodule
 f61416fef691d9ba39a40df1ce72aa574f54c390 FMS (2019.01.03)
 6bc61df3c363f9134a46439ff4a5a4a803daafb1 FV3 (heads/develop)
 9d05172b711f4ab5d6f978dbe575bd67a681b55a NEMS (heads/develop)
 96e3f3a8fa0389a4b110b0fa23e7a414f6d92038 WW3 (6.07.1-50-g96e3f3a8)
 ffdd19bc6c1df747394b7e9958a76238fcd44242 stochastic_physics
(ufs-v1.0.0-70-gffdd19b)

Remove aerosol from checkout

KateFriedman-NOAA committed:
Update post.sh UPP JJOB script name to submit

lgannoaa committed:
Making J-Job naming change accourding to code manager.
Remove temp files

jkhender committed:
modifications to python scripts
   remove aeroic task from setup_workflow_fcstonly.py
   delete setup_workflow_fcstonly_noaeroic.py
   add setup_workflow_fcstonly_aeroic.py

KateFriedman-NOAA committed:
Update WAFS jobs/rocoto scripts to use new JJOB names

KateFriedman-NOAA committed:
Update config.awips for newly named JJOB scripts

KateFriedman-NOAA committed:
Merge pull request #139 from NOAA-EMC/feature/gfsv16b_jobname

Update script names to match ecflow convention

Commit:
Issue #94 add /fakedbn to run DBN_alerts

RussTreadon-NOAA committed:
Issue #1:  update vrfy.sh to submit vsdb processing as separate job (only on WCOSS_DELL_P3)

KateFriedman-NOAA committed:
Update gfswafs job to run with loop over fcsthrs

KateFriedman-NOAA committed:
Update GLDAS tag to gldas_gfsv16_release.v1.10.0

lgannoaa committed:
Bring change from upstream_feature_gfsv16b

lgannoaa committed:
The EMC realtime parallel does not use operational job settings. Ecflow job card roll back the setting from module_used_gfs-16_job google sheet document.

lgannoaa committed:
Code manager indicated all wafs jobs wall clock is 30 mins.
Code manager indicated job card for scripts/gfs/atmos/gempak/jgfs_atmos_pgrb2_spec_gempak.ecf need to be changed

JessicaMeixner-NOAA committed:
Merge remote-tracking branch 'EMC/feature/gfsv16b' into feature/gfsv16b-wave-byhr

Conflicts:
	parm/config/config.resources

RussTreadon-NOAA committed:
Issue #1:  update parm/config.vrfy to define VSDBJOBSH (used by jobs/rocoto/vrfy.sh)

GuangPingLou-NOAA committed:
Merge remote-tracking branch 'upstream/feature/gfsv16b' into feature/gfsv16b

lgannoaa committed:
Merge remote-tracking branch 'upstream/feature/gfsv16b' into feature/gfsv16b

JessicaMeixner-NOAA committed:
Merge remote-tracking branch 'EMC/feature/gfsv16b' into feature/gfsv16b-wave-byhr

lgannoaa committed:
Jobs were tested with PDY 20200925, code managers from post, gempak, wave, and post process certified the test run result.

This merge included an update from high watermark testing (tested by gfs team using devonprod) results impact the following jobs:
jgfs_atmos_analysis.ecf
jgfs_forecast.ecf
jgdas_atmos_analysis.ecf
jgdas_enkf_update.ecf
jgdas_enkf_ecen.ecf

JessicaMeixner-NOAA committed:
cleaning up the rearranged scripts

RussTreadon-NOAA committed:
Issue #1:  update name of ncdiag executable and source code directory to be conistent

lgannoaa committed:
Merge remote-tracking branch 'upstream/feature/gfsv16b' into feature/gfsv16b

Commit:
Issue #94 add native grids as default grids

Commit:
Issue #94 solving conflicts

jkhender committed:
- updated to use 28Sep develop ufs-weather-model, d021e7b0395ccac2b7a30b414b58a8c924d2784f
   f61416fef691d9ba39a40df1ce72aa574f54c390 FMS (2019.01.03)
   9e1ba7c7448a8d009f39b5588e9498a7dbab1c60 FV3 (heads/develop)
   9d05172b711f4ab5d6f978dbe575bd67a681b55a NEMS (heads/develop)
   96e3f3a8fa0389a4b110b0fa23e7a414f6d92038 WW3 (6.07.1-50-g96e3f3a8)
   ffdd19bc6c1df747394b7e9958a76238fcd44242 stochastic_physics (ufs-v1.0.0-70-gffdd19b)

 - changed compilation options in build_fv3.sh
 - removed fv3gfs.fd_jkh directory since changes are already in develop branch
 - updated getic script to retrieve files after 00Z 26Feb20 from mass store with prefix name of 'com'

KateFriedman-NOAA committed:
Merge pull request #140 from lgannoaa/feature/gfsv16b

Update ecflow scripts for GFSv16

KateFriedman-NOAA committed:
Issue #1 - update WAFS tag to gfs_wafs.v6.0.8

Commit:
Issue #94 fix for failing silently

Commit:
Merge remote-tracking branch 'upstream/feature/gfsv16b' into feature/gfsv16_wave_prdgen

Issue #94 Adding error capture in the Jjobs

JessicaMeixner-NOAA committed:
updates for EE2 from waves

JessicaMeixner-NOAA committed:
last of EE2 changes

JessicaMeixner-NOAA committed:
update resources and trigger from 192->180

KateFriedman-NOAA committed:
Merge pull request #141 from RobertoPadilla-NOAA/feature/gfsv16_wave_prdgen

Updates to downstream wave jobs

JessicaMeixner-NOAA committed:
bug fix in exgfs_wave_post_pnt.sh

JessicaMeixner-NOAA committed:
fix resource time estimates

JessicaMeixner-NOAA committed:
Merge remote-tracking branch 'EMC/feature/gfsv16b' into feature/gfsv16b-wave-byhr

JessicaMeixner-NOAA committed:
reverting changes to configs that were not intented to be committed

KateFriedman-NOAA committed:
Issue #1 - updates for modules and small fixes

- fix to run ens_tracker without tclogg module in modulefiles/module_base.wcoss_dell_p3
- fix to sorc/syndat_qctropcy.fd/qctropcy.f for compile warning
- fix to gfswaveawipsbulls dependency in setup scripts
- update to bufr/11.3.0 from bufr/11.2.0 in modulefiles
- add override ability for POE for prep jobs in env/WCOSS_DELL_P3.env

KateFriedman-NOAA committed:
Merge pull request #144 from NOAA-EMC/feature/gfsv16b_updates

Issue #1 - updates for modules and small fixes

KateFriedman-NOAA committed:
Merge pull request #143 from JessicaMeixner-NOAA/feature/gfsv16b-wave-byhr

optimize wave post and EE2

GuangPingLou-NOAA committed:
issue #145 change dev path to prod for parallel netcdf modules

GuangPingLou-NOAA committed:
issue #145 change dev path to prod for parallel netcdf modules

GuangPingLou-NOAA committed:
issue #142 add 6 bufr stations for Thailand TMD

GuangPingLou-NOAA committed:
issue #142 add 6 bufr station data

GuangPingLou-NOAA committed:
issue #142 generate station i,j grid

GuangPingLou-NOAA committed:
issue #142 generate station i,j grid

GuangPingLou-NOAA committed:
issue #142 generate station i,j grid

KateFriedman-NOAA committed:
Issue #1 - adjust error handling in wave rocoto job scripts

KateFriedman-NOAA committed:
Merge branch 'feature/gfsv16b_updates' of https://github.com/NOAA-EMC/global-workflow into feature/gfsv16b_updates

KateFriedman-NOAA committed:
Merge pull request #148 from NOAA-EMC/feature/gfsv16b_updates

Updates to optimize wave post jobs

KateFriedman-NOAA committed:
Issue #1 - update component tags and modulefiles for nwtest lib updates, remove unneeded module load and modulefile from downstream wave job rocoto scripts

KateFriedman-NOAA committed:
Merge pull request #149 from GuangPingLou-NOAA/feature/gfsv16b

Add 6 bufr stations to the bufr sounding output

KateFriedman-NOAA committed:
Issue #1 - update GSI tag to gfsda.v16.0.0

KateFriedman-NOAA committed:
Issue #1 - remove unneeded line in vrfy.sh and update link_fv3gfs.sh for UFS_UTILS execs

KateFriedman-NOAA committed:
Issue #1 - config updates from real-time parallel

KateFriedman-NOAA committed:
Merge pull request #150 from NOAA-EMC/feature/gfsv16b_updates

nwtest module library and tag updates

KateFriedman-NOAA committed:
Issue #1 - update g2tmpl module load in modulefiles/module_base.wcoss_dell_p3

KateFriedman-NOAA committed:
Issue #1 - update prep job resources

KateFriedman-NOAA committed:
Issue #1 - update config.fv3 based on real-time parallel

KateFriedman-NOAA committed:
Issue #1 - move ABIBF, AHIBF, and HDOB pointers into RUN_ENVIR=emc block

KateFriedman-NOAA committed:
Issue #1 - remove hardcoded POE and BACK values from config.prep

KateFriedman-NOAA committed:
Issue #1 - return POE=YES and BACK=off setting for prep on WCOSS_DELL_P3

JessicaMeixner-NOAA committed:
for rocoto add a dependency to wavepostpnt job on wavepostbndpnt
so that both jobs will not run at the same time which will
slow both jobs down. This is the reason for the dependency, otherwise
there is not a "true" dependency between the jobs

KateFriedman-NOAA committed:
Merge pull request #152 from JessicaMeixner-NOAA/feature/addwavedependency

for rocoto add a dependency to wavepostpnt job on wavepostbndpnt

KateFriedman-NOAA committed:
Issue #1 - remove unneeded DMPDIR and ICSDIR from config.base.nco.static

KateFriedman-NOAA committed:
Issue #1 - remove POE/BACK block from config.prep and set POE=YES/BACK=off as defaults in env/WCOSS_DELL_P3.env prep section

KateFriedman-NOAA committed:
Issue #1 - update WAFS tag to gfs_wafs.v6.0.9

KateFriedman-NOAA committed:
Temporarily peg GSI checkout to hash of release branch

JessicaMeixner-NOAA committed:
add a dependency for the wavepostpnt on wavepostbndpnt for just gfs
as this job does not exist for gdas

lgannoaa committed:
Modify module for each job to match implementation package change
Modify two wafs jobs trigger
Modify wall clock and resource for running jobs in NCO
Modify obsproc package location

JessicaMeixner-NOAA committed:
updates to parm to reduce the number of wave variables
changes to the config so that wave models are interpolated
to the multi_1 masked files for the regional output grids

KateFriedman-NOAA committed:
Merge pull request #153 from JessicaMeixner-NOAA/bugfix/wavedependency

update wave post pnt dependency

KateFriedman-NOAA committed:
Issue #1 - update to fbwndgfs modulefiles for WCOSS-Dell and WCOSS-Cray

CoryMartin-NOAA committed:
Change config.resources for eobs for low resolution cases

KateFriedman-NOAA committed:
Issue #1 - update gfs_util modulefiles

KateFriedman-NOAA committed:
Merge pull request #154 from CoryMartin-NOAA/bugfix/eobs_resources

Change config.resources for eobs for low resolution cases

KateFriedman-NOAA committed:
Issue #1 - update config.fv3 and config.resources with v16rt2 values

KateFriedman-NOAA committed:
Issue #1 - update WAFS tag to gfs_wafs.v6.0.10 and change WAFS job dependencies

lgannoaa committed:
Merge remote-tracking branch 'upstream/feature/gfsv16b' into feature/gfsv16b

lgannoaa committed:
Modify two wafs jobs trigger as:
  jgfs_atmos_wafs_grib2          trigger ../../post/jgfs_atmos_post_f000 == complete
  jgfs_atmos_wafs_grib2_0p25     trigger ../../post/jgfs_atmos_post_f036 == complete

KateFriedman-NOAA committed:
Merge pull request #155 from lgannoaa/feature/gfsv16b

Further ecflow updates

KateFriedman-NOAA committed:
Issue #1 - update link_fv3gfs.sh to point to newly frozen fix_nco_gfsv16 FIX_DIR

lgannoaa committed:
Modify gfs/gdas post job to 20 minutes in wall clock.

KateFriedman-NOAA committed:
Merge pull request #156 from lgannoaa/feature/gfsv16b

Modify gfs/gdas post job to 20 minutes in wall clock.

JessicaMeixner-NOAA committed:
updates to add glo_30m to the created grib files for waves
for awips processing

JessicaMeixner-NOAA committed:
Merge remote-tracking branch 'EMC/feature/gfsv16b' into feature/gfsv16b-wavegrids

KateFriedman-NOAA committed:
Merge pull request #157 from JessicaMeixner-NOAA/feature/gfsv16b-wavegrids

Adding wave grids to grib interpolation

Commit:
Issue #94 producing awips files with masks and deleting wmo headers for arctic ocean

Commit:
Merge remote-tracking branch 'upstream/feature/gfsv16b' into feature/gfsv16_wave_prdgen

Generating awips files with masks and elimination wmo headers for Arctic Ocean

KateFriedman-NOAA committed:
Merge pull request #158 from RobertoPadilla-NOAA/feature/gfsv16_wave_prdgen

Feature/gfsv16 wave prdgen

KateFriedman-NOAA committed:
GFSv16 package changes from feature/gfsv16b before final wave updates

KateFriedman-NOAA committed:
Merge remote-tracking branch 'origin/feature/gfsv16b' into release/gfsv16.0.0

KateFriedman-NOAA committed:
Issue #1 - adding release notes for GFSv16

KateFriedman-NOAA committed:
Issue #1 - adjust WAFS dependencies to wait for f036 post output

KateFriedman-NOAA committed:
Sync merge with release/gfsv16.0.0 branch

lgannoaa committed:
Merge remote-tracking branch 'origin/feature/gfsv16b' into feature/ccpp

KateFriedman-NOAA committed:
Add wave env updates into ORION.env, increase waveinit tasks, and update resources for analysis and efcs walltimes

KateFriedman-NOAA committed:
Update .gitignore for script renaming and removed external files

KateFriedman-NOAA committed:
Issue #1 - update FV3 tag to GFS.v16.0.14 for Hera/Orion build support

KateFriedman-NOAA committed:
Issue #1 - new FV3 tag that builds on Hera/Orion and update for building/running high res system on Hera

KateFriedman-NOAA committed:
Issue #1 - save GFSv16 hand-off resource settings into new static nco configs

KateFriedman-NOAA committed:
Issue #1 - add FDATE to config.base

KateFriedman-NOAA committed:
Issue #1 - add parm mon folder to ignore list and move parm section of ignore list up

KateFriedman-NOAA committed:
Issue #1 - pull in corrected npe_eobs values in config.resources

KateFriedman-NOAA committed:
Add UPP netcdf module library load to HERA.env to resolve runtime netcdf version mismatch

KateFriedman-NOAA committed:
Merge remote-tracking branch 'origin/release/gfsv16.0.0' into port2orion

* origin/release/gfsv16.0.0:
  Issue #1 - pull in corrected npe_eobs values in config.resources
  Issue #1 - update FV3 tag to GFS.v16.0.14 for Hera/Orion build support

KateFriedman-NOAA committed:
Issue #1 - increase C768 npe_eobs to 200 for Hera nodes with less memory

KateFriedman-NOAA committed:
Issue #1 - resource updates from low res testing

- set io_layout to "1,1" for low res gfs in config.fcst
- set npe_wav[_gfs] to 140 for all resolutions, same wave grid
- increase nth_fv3 to 4 for C192 and C384
- set smaller walltime for gfsfcst when resolutions less than C768

KateFriedman-NOAA committed:
Issue #1 - fix firstcyc queue bug in workflow_utils.py on WCOSS

KateFriedman-NOAA committed:
Issue #1 - add wtime_fcst_gfs time of 4hrs for C384

KateFriedman-NOAA committed:
Fix wrong DATE in new if-block in config.anal for cold start checking

KateFriedman-NOAA committed:
Issue #1 - increase wavepostbndpnt and wavepostpnt walltimes to give more time on Hera

lgannoaa committed:
Cycled warm start run using develop ufs-weather-model branch and feature/ccpp using merged branch on the 10/13 (4da0f) with feature/gfsv16b (31563).
CPP_SUITE="FV3_GFS_v16beta"
build_fv3 option: CCPP=Y 32BIT=Y SUITES=FV3_GFS_v15,FV3_GSD_v0,FV3_GSD_noah,FV3_GFS_v16beta
wave is turned off for this test because it's for atmospheric physics and comparison.

lgannoaa committed:
Merge remote-tracking branch 'upstream/port2orion' into feature/ccpp

KateFriedman-NOAA committed:
Issue #1 - update WAFS tag to gfs_wafs.v6.0.12 for removal of in-cloud turbulence per AWC

KateFriedman-NOAA committed:
Rename Release_Notes.gfs.v16.0.0.txt to Release_Notes.gfs.v16.0.0.md

KateFriedman-NOAA committed:
Issue #1 - adjust epos groups for DOIAU/DOIAU_ENKF=NO

KateFriedman-NOAA committed:
Issue #1 - update anal, eobs, and eupd resources for low res

KateFriedman-NOAA committed:
Issue #1 - update WAFS tag to gfs_wafs.v6.0.13

KateFriedman-NOAA committed:
Revert epos change in setup_workflow.py

yangfanglin committed:
modified:   jobs/JGFS_ATMOS_POSTSND and jobs/rocoto/postsnd.sh to
            remove redundant variables in the two scripts and make them
            work for both EMC and NCO parallels.

yangfanglin committed:
Merge branch 'release/gfsv16.0.0' of https://github.com/NOAA-EMC/global-workflow into release/gfsv16.0.0

KateFriedman-NOAA committed:
Issue #1 - update WAFS tag to gfs_wafs.v6.0.14 and update dumpjb version to 5.1.0

KateFriedman-NOAA committed:
Reduce C384 nth_fv3 to 1 in config.fv3

KateFriedman-NOAA committed:
Add FDATE calculation to setup scripts and change FDATE in to parsed value in config.base.emc.dyn

KateFriedman-NOAA committed:
Set nth_fcst to 4 for C384 on Hera to handle less memory

KateFriedman-NOAA committed:
Increase gfsfcst walltime for C192

KateFriedman-NOAA committed:
Tie DOIAU_ENKF to DOIAU and add DOIAU check for IAU_OFFSET and IAU_FHROT in config.base

KateFriedman-NOAA committed:
Set nth_fcst to 4 for C384 deterministic

KateFriedman-NOAA committed:
Hera updates for OUTPUT_FILETYPES and resources

KateFriedman-NOAA committed:
Merge remote-tracking branch 'origin/release/gfsv16.0.0' into port2orion

* origin/release/gfsv16.0.0:
  Issue #1 - update WAFS tag to gfs_wafs.v6.0.14 and update dumpjb version to 5.1.0
  modified:   jobs/JGFS_ATMOS_POSTSND and jobs/rocoto/postsnd.sh to             remove redundant variables in the two scripts and make them             work for both EMC and NCO parallels.
  Issue #1 - update WAFS tag to gfs_wafs.v6.0.13
  Rename Release_Notes.gfs.v16.0.0.txt to Release_Notes.gfs.v16.0.0.md
  Issue #1 - update WAFS tag to gfs_wafs.v6.0.12 for removal of in-cloud turbulence per AWC

KateFriedman-NOAA committed:
Change wavepostbndpnt to wait for fcst to end

KateFriedman-NOAA committed:
Sync merge with develop to resolve conflicts

KateFriedman-NOAA committed:
Merge pull request #174 from NOAA-EMC/port2orion

GFSv16 release hand-off state, Orion support, and additional low res R&D updates

lgannoaa committed:
As of 11/12/2020, this package is tested with a C768 cycled (6 cycle) run on Dell system using FV3_GFS_v16beta (imp_physics=11). The wave components are turned off.
Ready to merge back to the development.
The expdir and log files are available on HPSS:
/NCEPDEV/emc-global/1year/Lin.Gan/WCOSS_DELL_P3/feature_ccpp/FV3_GFS_v16beta/FV3_GFS_v16beta-LOG.tar
/NCEPDEV/emc-global/1year/Lin.Gan/WCOSS_DELL_P3/feature_ccpp/FV3_GFS_v16beta/FV3_GFS_v16beta-EXPT.tar
File parm/config/config.resources has been modified to fix EUPD job issue in job card resource.
File sorc/build_fv3.sh and sorc/link_fv3gfs.sh changed to clear up development remark.
Removed a developer directory that is not required for FV3_GFS_v16beta confiruration.

lgannoaa committed:
Merge remote-tracking branch 'upstream/develop' into feature/ccpp

lgannoaa committed:
Changes to be committed:
       modified:   jobs/rocoto/fcst.sh
       deleted:    modulefiles/module_base.wcoss_dell_p3_fcst
       deleted:    parm/config/config.aeroic
       modified:   parm/config/config.base.emc.dyn
       modified:   parm/config/config.resources
       modified:   sorc/build_fv3.sh
       modified:   sorc/checkout.sh
       modified:   sorc/link_fv3gfs.sh
       deleted:    ush/load_fv3gfs_modules_fcst.sh
       deleted:    ush/rocoto/setup_workflow_fcstonly_aeroic.py

lgannoaa committed:
deleted:    jobs/rocoto/aeroic.sh

lgannoaa committed:
Remove GSD suite file and aero IC extract util

lgannoaa committed:
sync config.base.emc.dyn and config.resources with development

yangfanglin committed:
modified:   parm/config/config.base.emc.dyn
  parm/config/config.base.nco.static
  scripts/exglobal_forecast.sh
  sorc/build_all.sh
  sorc/build_fv3.sh
  sorc/checkout.sh
 to add the option to check out ufs_weather_model and run with CCPP phsyics

lgannoaa committed:
Merge with v16ccpp
       modified:   parm/config/config.base.emc.dyn
       modified:   parm/config/config.resources
       modified:   scripts/exglobal_forecast.sh
       modified:   sorc/build_fv3.sh
       modified:   sorc/checkout.sh

lgannoaa committed:
update eupd resource

lgannoaa committed:
Merge remote-tracking branch 'upstream/v16ccpp' into feature/ccpp

lgannoaa committed:
Remove development configuration from config.nsst

yangfanglin committed:
modified:   ../scripts/exglobal_forecast.sh
modified:   build_all.sh  build_fv3.sh to automatically detect model version, CCPP vs IPD

lgannoaa committed:
Modified build_all.sh checkout.sh partial_build.sh to allow switch turn on/off ccpp option
Turn on CCPP usage:
  checkout.sh -c
  build_all.sh -c
Turn off CCPP usage:
  checkout.sh
  build_all.sh

lgannoaa committed:
Clean up development code from build_all

yangfanglin committed:
modified:   build_fv3.sh

yangfanglin committed:
modified:   exglobal_forecast.sh to add
  min_lakeice  = ${min_lakeice:-"0.15"}
  min_seaice   = ${min_seaice:-"0.15"}

yangfanglin committed:
bug fix exglobal_forecast.sh

yangfanglin committed:
modified:   ../parm/config/config.fcst and   exglobal_forecast.sh
1. use iovr for model after https://github.com/NCAR/ccpp-physics/pull/514
   and iovr_sw and iovr_lw for older version of the model, controled by RUN_CCPP
2. remove the if block and use atmos_model_nml to turn on/off the CCPP option

yangfanglin committed:
modified checkout.sh to check out a hash instead of head of the develop branch

yangfanglin committed:
modified:   ../scripts/exglobal_forecast.sh

lgannoaa committed:
Modify exglobal_forecast.sh to create namelist in runable sequence.

lgannoaa committed:
Merge remote-tracking branch 'upstream/v16ccpp' into feature/ccpp
Pull in the iovr=3 in config.fcst

lgannoaa committed:
As requested, using 2e25df5fe952d27355ed58963148f46b82565469 for ufs-weather-model.

lgannoaa committed:
scripts/exglobal_forecast.sh Merge with v16ccpp

lgannoaa committed:
Modify build_all.sh
while getopts "oc" option
to
while getopts "c" option

KateFriedman-NOAA committed:
Fix missing COMROOT setting on Orion

lgannoaa committed:
Merge pull request #1 from KateFriedman-NOAA/ccpp

Fix missing COMROOT setting for Orion

KateFriedman-NOAA committed:
Adjust efcs walltime and eupd thread value on Orion

lgannoaa committed:
Merge pull request #2 from KateFriedman-NOAA/ccpp

Adjust efcs walltime and eupd thread value on Orion

KateFriedman-NOAA committed:
Merge pull request #186 from lgannoaa/feature/ccpp

Initial support for CCPP.

Commit:
gfsv16 physics; script updates

Commit:
update IC scripts

JessicaMeixner-NOAA commented on jobs/JGLOBAL_FORECAST:
```diff
@@ -59,6 +59,11 @@ export pgmerr=errfile
 ##############################################
 export CDATE=${CDATE:-${PDY}${cyc}}
 export CDUMP=${CDUMP:-${RUN:-"gfs"}}
+
```
Should this have if statements as in the develop branch: https://github.com/NOAA-EMC/global-workflow/blob/develop/jobs/JGLOBAL_FORECAST#L73-L79

jkuang-exploro replied:
I think it should be OK either way. These variables already appear in JJOB so I'm not sure why they are here at all. But I included them just in case there's some other concerns that I'm not yet aware of. 

These scripts probably need a extensive clean up of excess variables, seems like.

JessicaMeixner-NOAA commented on ush/forecast_postdet.sh:
```diff
@@ -537,12 +559,12 @@ WW3_postdet()
     HMS="$(echo $YMDH | cut -c9-10)0000"
       for wavGRD in ${waveGRD} ; do
-        eval $NLN $datwave/${wavprfx}.out_grd.${wavGRD}.${YMD}.${HMS} $DATA/${YMD}.${HMS}.out_grd.${wavGRD}
+        eval $NLN $datwave/${wavprfx}.out_grd.${wavGRD}.${YMD}.${HMS} ${YMD}.${HMS}.out_grd.${wavGRD}
```
why was DATA removed?

jkuang-exploro replied:
It should be preserved. Let me fix it.

JessicaMeixner-NOAA commented on ush/forecast_postdet.sh:
```diff
@@ -551,13 +573,11 @@ WW3_postdet()
     YMD=$(echo $YMDH | cut -c1-8)
     HMS="$(echo $YMDH | cut -c9-10)0000"
-      eval $NLN $datwave/${wavprfx}.out_pnt.${waveuoutpGRD}.${YMD}.${HMS} $DATA/${YMD}.${HMS}.out_pnt.${waveuoutpGRD}
+      eval $NLN $datwave/${wavprfx}.out_pnt.${waveuoutpGRD}.${YMD}.${HMS} ${YMD}.${HMS}.out_pnt.${waveuoutpGRD}
```
Why was DATA removed?

Commit:
model configure changes; preserve $DATA in postdet

Commit:
update /driver; /parm and /ecflow folder to match gfsv16

JessicaMeixner-NOAA commented:
@JianKuang-NOAA  I know that the forecast script has to be manually merged, however I'm confused by this last commit 45341b6 "update /driver; /parm and /ecflow folder to match gfsv16" shouldn't those changes be incorporated instead through 

git merge develop 

instead of a manual merge? 

Also- I thought this PR was going to be just the first part of changing the forecast script, then there would be a second update with namelist changes.   Then at some point the actual "git merge develop" and moving to use in-line post for the atm model?

jkuang-exploro commented:
@JessicaMeixner-NOAA /driver; /parm and /ecflow are top directories containing files that won't be called in a CROW-based workflow; thus we have never use them, nor will in the future. For this reason, I simply update them by fully matching develop branch, largely for documentation and code management purpose.

As you know some files may needs more attention than others; so my approaches differ.

JessicaMeixner-NOAA commented:
@JianKuang-NOAA if they are not used by CROW, all the more reason for retain their proper history by updating those files through "git merge develop" instead of by hand.

jkuang-exploro commented:
@JessicaMeixner-NOAA That's what I've done for those directories.

JessicaMeixner-NOAA commented:
How can you maintain their git histories if you copied them from the other branch instead of doing "git merge"? 

> @JessicaMeixner-NOAA That's what I've done for those directories.

jkuang-exploro commented:
@JessicaMeixner-NOAA Oh. I thought they were equivalent. I will start over.

JessicaMeixner-NOAA commented:
Instead of restarting, can we just revert this last commit?

jkuang-exploro commented:
@JessicaMeixner-NOAA Yes. Will do.

Commit:
Revert "update /driver; /parm and /ecflow folder to match gfsv16"

This reverts commit 45341b6ae35e125334b34bbdda602cb13808e8ef.

JessicaMeixner-NOAA commented:
@JianKuang-NOAA thanks for reverting the last commit.  I ran a test with your updates and the forecast job failed.  I think it might be a disk space or user error so I'm rerunning and will let you know.

JessicaMeixner-NOAA commented:
@JianKuang-NOAA my early error seems to have gone away however, now all of the grib2 files for the atmosphere are empty.  I suspect this is because only the forecast script has been moved to the new com directory structure but the atmosphere post scripts have not been updated to use the new com directory structure.  It might not end up being possible to do all of this in separate parts after all.

jkuang-exploro commented:
@JessicaMeixner-NOAA OK let me check.

JessicaMeixner-NOAA commented:
From my tests: 
COM from this PR branch: /work/noaa/marine/jmeixner/p6pr194/COMROOT/test01/
COM from top of coupled-crow: /work/noaa/marine/jmeixner/p6dev/COMROOT/test01/
that I was comparing

Commit:
Merge branch 'develop' into feature/gfsv16intocoupled

Conflicts:
	.gitignore
	Externals.cfg
	README.md
	ecflow/ecf/versions/obsproc_global.ver
	env/JET.env
	env/ORION.env
	jobs/JGDAS_GEMPAK
	jobs/JGFS_AWIPS_20KM_1P0DEG
	jobs/JGFS_AWIPS_G2
	jobs/JGFS_GEMPAK
	jobs/JGFS_GEMPAK_META
	jobs/JGFS_GEMPAK_NCDC_UPAPGIF
	jobs/JGFS_PGRB2_SPEC_GEMPAK
	jobs/JGFS_PGRB2_SPEC_NPOESS
	jobs/JGFS_POSTSND
	jobs/JGLOBAL_EMCSFC_SFC_PREP
	jobs/JGLOBAL_FORECAST
	jobs/JGLOBAL_WAVE_INIT
	jobs/JGLOBAL_WAVE_POST_PNT
	jobs/JGLOBAL_WAVE_POST_SBS
	jobs/JGLOBAL_WAVE_PREP
	jobs/rocoto/arch.sh
	jobs/rocoto/earc.sh
	jobs/rocoto/prep.sh
	jobs/rocoto/waveinit.sh
	jobs/rocoto/wavepostsbs.sh
	jobs/rocoto/waveprep.sh
	modulefiles/fv3gfs/enkf_chgres_recenter.wcoss_dell_p3
	modulefiles/fv3gfs/gaussian_sfcanl.hera
	modulefiles/fv3gfs/gaussian_sfcanl.wcoss_dell_p3
	modulefiles/gfs_bufr.orion
	modulefiles/gfs_bufr.wcoss_dell_p3
	modulefiles/gfs_fbwndgfs.orion
	modulefiles/module_base.hera
	modulefiles/module_base.orion
	modulefiles/module_base.wcoss_dell_p3
	modulefiles/modulefile.fv3nc2nemsio.orion
	modulefiles/modulefile.regrid_nemsio.orion
	modulefiles/modulefile.storm_reloc_v6.0.0.orion
	parm/config/config.anal
	parm/config/config.base.emc.dyn
	parm/config/config.base.nco.static
	parm/config/config.efcs
	parm/config/config.eobs
	parm/config/config.epos
	parm/config/config.fcst
	parm/config/config.gempak
	parm/config/config.post
	parm/config/config.postsnd
	parm/config/config.prep
	parm/config/config.prepbufr
	parm/config/config.resources
	parm/config/config.vrfy
	parm/config/config.wave
	parm/config/config.wavepostsbs
	parm/config/config.waveprep
	scripts/exgdas_nawips.sh.ecf
	scripts/exgdas_tropc.sh.ecf
	scripts/exgempak_gfs_gif_ncdc_skew_t.sh.ecf
	scripts/exgfs_fax.sh.ecf
	scripts/exgfs_nawips.sh.ecf
	scripts/exgfs_postsnd.sh.ecf
	scripts/exgfs_wave_init.sh
	scripts/exgfs_wave_post_gridded_sbs.sh
	scripts/exgfs_wave_post_pnt.sh
	scripts/exgfs_wave_prep.sh
	scripts/exglobal_fcst_nemsfv3gfs.sh
	scripts/exglobal_grib2_special_npoess.sh.ecf
	scripts/exgoes_nawips.sh.ecf
	scripts/exnawips.sh.ecf
	sorc/build_all.sh
	sorc/build_tropcy_NEMS.sh
	sorc/checkout.sh
	sorc/enkf_chgres_recenter.fd/interp.f90
	sorc/fv3gfs_build.cfg
	sorc/link_fv3gfs.sh
	sorc/machine-setup.sh
	sorc/partial_build.sh
	ush/gfs_bufr.sh
	ush/hpssarch_gen.sh
	ush/load_fv3gfs_modules.sh
	ush/rocoto/setup_workflow.py
	ush/wave_grib2_sbs.sh
	ush/wave_grid_interp_sbs.sh
	ush/wave_outp_spec.sh
	ush/wave_prnc_cur.sh
	ush/wave_prnc_ice.sh
	ush/wave_tar.sh

jkuang-exploro commented:
@JessicaMeixner-NOAA I am ready to push another commit containing resolutions of "git merge develop". Do you want me to hold up that?

JessicaMeixner-NOAA commented:
@JianKuang-NOAA No - go for it.  As is this PR cannot be merged, hopefully the update you will make will resolve the issues I saw.

jkuang-exploro commented:
@JessicaMeixner-NOAA I am running into an error in my test case and it is likely related to WW3:

                    *** WAVEWATCH III Multi-grid shell ***
               =================================================


EXTCDE MPI_ABORT, IEXIT=  2002

slurmstepd: error: *** STEP 959619.0 ON Orion-24-24 CANCELLED AT 2020-12-02T15:11:52 ***

EXTCDE MPI_ABORT, IEXIT=  2002


EXTCDE MPI_ABORT, IEXIT=  2002

jkuang-exploro commented:
Above is the log. Here are full information of this case:

EXPDIR: /work/noaa/marine/Jian.Kuang/expdir/casetest
ROTDIR: /work/noaa/marine/Jian.Kuang/comroot/casetest
DATA dir for forecast job: /work/noaa/marine/Jian.Kuang/dataroot/casetest/2013040100/gfs/fcst.62259

Would you please take a look at them when possible? I will keep investigate on my side as well.

JessicaMeixner-NOAA commented:
@JianKuang-NOAA  thanks for the info - I'll look into it.

JessicaMeixner-NOAA commented:
@JianKuang-NOAA can you check in your local changes you have so far?   I'll need them to make a few more changes to the wave init and prep jobs & scripts to get them to work.  Otherwise, I'll just redo them and make a PR back to this branch when I'm done.  Let me know your preference.

jkuang-exploro commented:
Of course. I will do it right now.

Commit:
configuration update

jkuang-exploro commented:
@JessicaMeixner-NOAA I've pushed all my local changes, mostly in the /workflow directory, for the additional config.wave* files and JJOB of waveinit.

JessicaMeixner-NOAA commented:
@JianKuang-NOAA I have fixed the first problem and now am having others that will have to be sorted out by syncing the configs from the waves from the develop branch system to the crow-configure system (+ update values to be appropriate for the benchmark configuration)  I'm putting my updates here: https://github.com/JessicaMeixner-NOAA/global-workflow/tree/coupled/devmerge and will make a pr when the wave parts are ready.   It'll probably take another day or so.

jkuang-exploro commented:
@JessicaMeixner-NOAA OK I will test run your other PRs now.

JessicaMeixner-NOAA commented:
@JianKuang-NOAA I have made updates to have the branch match more closely with the develop branch (mostly from files that did not get deleted for some reason with the merge) and other updates related to the waves in my branch here: https://github.com/JessicaMeixner-NOAA/global-workflow/tree/coupled/devmerge  The ATM post still does not work due to the com restructuring I believe.     If PR #199 gets merged first, I can merge that in here before making a PR with my changes to your branch feature/gfsv16intocoupled if you want.

jkuang-exploro commented:
@JessicaMeixner-NOAA OK. The test case finished just now for PR199. Let me merge it.

Commit:
Merge remote-tracking branch 'origin/feature/coupled-crow' into feature/gfsv16intocoupled

Conflicts:
	ush/forecast_postdet.sh
	ush/forecast_predet.sh

Commit:
preserve /atmos level COM directory

Commit:
typo fix

JessicaMeixner-NOAA committed:
Coupled/devmerge (#203)

* update to waves scripts for merging with develop
- synced with develop where necessary and left changes in
prep and gridded post for coupled-crow updates that should remain

* removing ecflow scripts that are not in develop branch

* removing files from branch that do not appear in develop branch
and were not added for coupling in the coupled-crow branch

* updating sorc and ush files to match dev when these are not files
changes intentionally in coupled-crow branch and therefore should
match develop branch

* updating files to match develop that were not matching through
merge but should match develop branch

* cleaning up evn files after merge with develop

* cleaning up differences in jobs/rocoto folder from develop branch
as it does not seem that the changes here were made intentionally
in feature/coupled-crow so this brings us insync with the develop branch

* resolving differences between develop and this branch in terms
of wave variables for jobs/JGLOBAL_FORECAST

* updating module file to match develop

* updates for wave model so that the forecast runs
-- atm post still does not work

* fix typo

KateFriedman-NOAA committed:
Fixes for issue #202 (FINDDATE) and issue #208 (postsnd.sh permissions)

JessicaMeixner-NOAA committed:
coupled: updates for merging checkout, build and link with develop branch (#206)

* update to waves scripts for merging with develop
- synced with develop where necessary and left changes in
prep and gridded post for coupled-crow updates that should remain

* removing ecflow scripts that are not in develop branch

* removing files from branch that do not appear in develop branch
and were not added for coupling in the coupled-crow branch

* updating sorc and ush files to match dev when these are not files
changes intentionally in coupled-crow branch and therefore should
match develop branch

* updating files to match develop that were not matching through
merge but should match develop branch

* cleaning up evn files after merge with develop

* cleaning up differences in jobs/rocoto folder from develop branch
as it does not seem that the changes here were made intentionally
in feature/coupled-crow so this brings us insync with the develop branch

* resolving differences between develop and this branch in terms
of wave variables for jobs/JGLOBAL_FORECAST

* updating module file to match develop

* updates for wave model so that the forecast runs
-- atm post still does not work

* fix typo

* updating .gitignore to match develop a little better

* cleaning up differences between checkout and link with merging to develop

* updates for build and checkout scripts to unify with develop
and simplify build_all to add an option for coupling

* fix typo in build all script

* updating .gitignore for linked files missing from list

* add fix files for ocean post

* changing -p to -c option and adding coupled to the ccpp -c

Commit:
configuration and minor script update for inline post

jkuang-exploro commented:
Please review this PR once again. 

Test routine is identical with previous practice. Inline post is now a default setting, which user could turn off in the case file. Workflow layout will now change based on using inline post (default) or not.

JessicaMeixner-NOAA commented:
@JianKuang-NOAA I will start my review of this PR... before I do since #210 is also ready to go should we merge that in first to feature/coupled-crow and pull those updates back in?  The only issue with that PR not being merged already is that it wasn't reviewed w/someone who has write permissions. 

Also another question: This completes the merge of everything from develop into feature/coupled-crow including the use of in-line post.  Does this also update the namelist changes that needed to be merged by hand into input.nml?

jkuang-exploro commented:
@JessicaMeixner-NOAA OK I will review PR#210. And yes this PR contains namelists updates.

Commit:
Merge remote-tracking branch 'origin/feature/coupled-crow' into feature/gfsv16intocoupled

Conflicts:
	sorc/checkout.sh

JessicaMeixner-NOAA commented:
@JianKuang-NOAA  maybe this is a one-off weird issue, but I just had this happen where the verify job is triggered very early somehow: 

```
       CYCLE                    TASK                       JOBID               STATE         EXIT STATUS     TRIES      DURATION
================================================================================================================================
201304010000     ics.jgfs_emc_mom6ic                      985328           SUCCEEDED                   0         1          40.0
201304010000       gfs.prep.waveinit                      985330           SUCCEEDED                   0         1          19.0
201304010000       gfs.prep.waveprep                      985429           SUCCEEDED                   0         1           3.0
201304010000    gfs.forecast.highres                      985529              QUEUED                   -         0           0.0
201304010000       gfs.ocnpost.p_000                           -                   -                   -         -             -
201304010000       gfs.ocnpost.p_024                           -                   -                   -         -             -
201304010000            gfs.wavepost                           -                   -                   -         -             -
201304010000         gfs.wavepostpnt                           -                   -                   -         -             -
201304010000       gfs.jgfs_emc_vrfy                      985329           SUCCEEDED                   0         1           4.0
201304010000    archive.jgfs_archive                           -                   -                   -         -             -
201304010000                   final                           -                   -                   -         -             -
```

jkuang-exploro commented:
@JessicaMeixner-NOAA Yeah I am aware of that. Now drafting a fix.

JessicaMeixner-NOAA commented:
@JianKuang-NOAA it didn't happen in my feature/coupled-crow branch test but I didn't see any changes from this branch that seemed like it was causing that so ?  Let me know if I should restart my test when the new code is ready.

Commit:
bug fix for vrfy job in layout file;
missing namelist variable fix

JessicaMeixner-NOAA commented:
@JianKuang-NOAA thanks for the fix. I'm starting a new test with the up to date code now.

JessicaMeixner-NOAA committed:
updates that are needed after merging

KateFriedman-NOAA committed:
Issue #201 - workaround for failing post000 job before hpc-stack solution

KateFriedman-NOAA committed:
Merge pull request #213 from NOAA-EMC/hotfixes

Hotfixes - issues #201, 202, 208

KateFriedman-NOAA committed:
Issue #189 - update ufs-weather-model hash

KateFriedman-NOAA committed:
Merge pull request #216 from NOAA-EMC/issue189

Issue #189 - update ufs-weather-model hash

jkuang-exploro committed:
Coupled/devmerge 3 (#219)

* update for making post work without inline post

* add in linking capabilities

* updating module file so that atm pos will work (#214)

Co-authored-by: JessicaMeixner-NOAA <Jessica.Meixner@noaa.gov>

Commit:
set inline_post off for default
file dependencies for post job triggering updated

jkuang-exploro commented:
@JessicaMeixner-NOAA I fixed the triggering issue for post jobs. And I set inline_post to be default off. The test on Orion is running well so far. Would you please test it on Hera?

JessicaMeixner-NOAA commented:
@JianKuang-NOAA that's great news.  After PR #218 gets merged into feature/coupled-crow and is merged into this branch I will run a test on hera.  Since that PR is ready to be merged and we'd have to re-do the test after merging that in here, I think it's better to do that first and skip a test.

Commit:
Merge remote-tracking branch 'origin/feature/coupled-crow' into feature/gfsv16intocoupled

Commit:
add new parameters into /default

Commit:
clean up excessive logic clause in namelist parsing scripts

JessicaMeixner-NOAA commented:
@JianKuang-NOAA thanks for merging in #218 and adding the fix by adding the variables to the defaults!  I noticed some additional commits on this branch for cleanup.  Are you ready for me to test this on hera or should I wait?

jkuang-exploro commented:
@JessicaMeixner-NOAA This PR is good to go per my test on Orion. Please test it on Hera and let me know how it goes. Thanks.

JessicaMeixner-NOAA commented:
@JianKuang-NOAA, I will start my test on hera now.

JessicaMeixner-NOAA commented:
@JianKuang-NOAA the run on hera failed the same way my run on orion failed.  See the log file 

/work/noaa/marine/jmeixner/p6pr194update/test01/COMROOT/test01/logs/2013040100/gfs.forecast.highres.log 

Important parts: 

```
0 + source /work/noaa/marine/jmeixner/p6pr194update/global-workflow/scripts/../ush/parsing_model_configure_FV3.sh
/work/noaa/marine/jmeixner/p6pr194update/global-workflow/scripts/../ush/parsing_model_configure_FV3.sh: line 76: syntax error near unexpected token `}'
/work/noaa/marine/jmeixner/p6pr194update/global-workflow/scripts/../ush/parsing_model_configure_FV3.sh: line 76: `}'
```

```
forrtl: severe (29): file not found, unit 101, file /work/noaa/marine/jmeixner/p6pr194update/test01/DATA/test01/2013040100/gfs/fcst.378149/model_configure

```

jkuang-exploro commented:
@JessicaMeixner-NOAA This is likely a typo. It is so strange that my test didn't capture it. Let me check.

JessicaMeixner-NOAA commented:
@JianKuang-NOAA sorry I can't do this through the normal review as there are so many files with changes it's hard to go through and mark them on the file.  But I just checked the diffs in the input.nml creation from the namelist options that are in ush/parsing_namelists_FV3.sh and we need to update the following to match the develop branch: 

These are the variables in the develop branch options that I found that did not match the defaults in this PR: 
adjust_dry_mass = ${adjust_dry_mass:-".true."} 
lheatstrg    = ${lheatstrg-".false."}
lgfdlmprad   = ${lgfdlmprad:-".false."}
lgfdlmprad   = ${lgfdlmprad:-".true."}
effr_in      = ${effr_in:-".false."}
do_ugwp      = ${do_ugwp:-".true."}  
do_tofd      = ${do_tofd:-".true."}
reiflag = ${reiflag:-"2"}

In the develop branch cplwav needs to default to false:   
cplwav       = ${cplwav:-".false."} 
not just be: 
cplwav       = ${cplwav}

The if  $CCPP_SUITE blocks starting on line 201 could be updated to just be one if block following the "if [ $RUN_CCPP = "YES" ]; then" if block in the develop branch starting on line 1092 

lines 234-5 should be removed: 
  iovr_lw      = ${iovr_lw:-"3"}
  iovr_sw      = ${iovr_sw:-"3"}

Line 486: 
&nam_sfcperts 
should instead be   
$nam_sfcperts_nml

JessicaMeixner-NOAA commented:
> @JessicaMeixner-NOAA This is likely a typo. It is so strange that my test didn't capture it. Let me check.

Are there local changes that did not get pushed?

JessicaMeixner-NOAA commented:
@JianKuang-NOAA Also doing a difference between the develop branch and this PR I saw that 
Files develop/sorc/build_ncep_post.sh and pr194/sorc/build_ncep_post.sh differ
which I don't think should necessarily be different. 

There were a couple of other files such as : 
Files develop/parm/config/config.gldas and pr194/parm/config/config.gldas differ
Files develop/env/HERA.env and pr194/env/HERA.env differ

which I believe is just there from recent commits to the develop branch.

jkuang-exploro commented:
@JessicaMeixner-NOAA I fixed the model_configure script issue which is a simple syntax error. I will fix those name list variables that you mentioned, together with it.

Commit:
Merge branch 'develop' into feature/gfsv16intocoupled
fix syntax error in model_configure script

Conflicts:
	sorc/checkout.sh

jkuang-exploro commented:
@JessicaMeixner-NOAA I think the RUN_CCPP block in the develop branch is not so clear because there are multiple CCPP suites and each of them may come with some additional name list variables. In other words, CCPP should be a selectable set instead of a  "true/false" variable. Indeed the RUN_CCPP block in the develop only takes care of the "FV3_GSD_v0" suite. Please correct me if I am wrong.

jkuang-exploro commented:
> @JianKuang-NOAA sorry I can't do this through the normal review as there are so many files with changes it's hard to go through and mark them on the file. But I just checked the diffs in the input.nml creation from the namelist options that are in ush/parsing_namelists_FV3.sh and we need to update the following to match the develop branch:
> 
> These are the variables in the develop branch options that I found that did not match the defaults in this PR:
> adjust_dry_mass = ${adjust_dry_mass:-".true."} 
> lheatstrg    = ${lheatstrg-".false."}
> lgfdlmprad   = ${lgfdlmprad:-".false."}
> lgfdlmprad   = ${lgfdlmprad:-".true."}
> effr_in      = ${effr_in:-".false."}
> do_ugwp      = ${do_ugwp:-".true."} 
> do_tofd      = ${do_tofd:-".true."}
> reiflag = ${reiflag:-"2"}
> 
> In the develop branch cplwav needs to default to false:   
> cplwav       = ${cplwav:-".false."}
> not just be:
> cplwav       = ${cplwav}
> 
> The if $CCPP_SUITE blocks starting on line 201 could be updated to just be one if block following the "if [ $RUN_CCPP = "YES" ]; then" if block in the develop branch starting on line 1092
> 
> lines 234-5 should be removed:
> iovr_lw = ${iovr_lw:-"3"}
> iovr_sw = ${iovr_sw:-"3"}
> 
> Line 486: 
> &nam_sfcperts
> should instead be   
> $nam_sfcperts_nml

Line 486 "nam_sfcperts_nml", is identical with the develop branch. I don't know what should I do?

Commit:
match default value of some namelist variables

JessicaMeixner-NOAA commented:
@JianKuang-NOAA It's my understanding that we need these for all CCPP suites, not just "FV3_GSD_v0"  In the develop branch I think the suite that was recently tested was FV3_GFS_v16beta and we're going to be using the coupled version of that for prototype 6.

jkuang-exploro commented:
> @JianKuang-NOAA It's my understanding that we need these for all CCPP suites, not just "FV3_GSD_v0" In the develop branch I think the suite that was recently tested was FV3_GFS_v16beta and we're going to be using the coupled version of that for prototype 6.

Yes. That's the reason why I think we shouldn't define CCPP as a YES/NO option. Instead we need to enable users to select suite by it's name.

JessicaMeixner-NOAA commented:
@JianKuang-NOAA those options are needed for any CCPP suite at this time.  If you want to keep it for a specific suite file, please add an if block for "FV3_GFS_v16beta_coupled" with all of the options in the develop branch in that if block. 

You are correct on the nam_sfcperts_nml, it is already how it is in the develop branch.  I am not sure what I saw before.

Commit:
name list update for CCPP suite "FV3_GFS_v16beta_coupled"
update CCPP suite selection in test case file
create prototype 5.1 case file with updated CCPP suite

Making each CCPP suite a selectable option is beneficial,
because this could handle future development when they
start to differ from each other.

JessicaMeixner-NOAA commented on workflow/cases/coupled_free_forecast.yaml:
```diff
@@ -34,7 +34,7 @@ case:
     DELTIM: 450
     SEEDLET: 10
-    CCPP_SUITE: FV3_GFS_v15p2_coupled
+    CCPP_SUITE: FV3_GFS_v16beta_coupled
```
@JianKuang-NOAA we aren't quite ready to jump to this yet.  can you please leave this as     
CCPP_SUITE: FV3_GFS_v15p2_coupled		
for the time being?

jkuang-exploro replied:
OK. I will revert this change.

JessicaMeixner-NOAA commented on workflow/cases/prototype_5.1.yaml:
```diff
@@ -0,0 +1,60 @@
+case:
```
This isn't needed at this time.  We are not running a prototype 5.1 yet.  The next one will be p6.  We'll make a case file for it when we are ready, but we are not there yet.

Commit:
change CCPP suite back to FV3_GFS_v15p2_coupled in case file

JessicaMeixner-NOAA commented:
I pulled the recent changes (up to 14a85cd3b7616cac213857b783bc44bce19b7d23) and tried again and the forecast still fails this time with: 
forrtl: severe (19): invalid reference to variable in NAMELIST input, unit -5, file Internal Formatted NML Read, line -1, position 16

Let me know when you are ready for me to run a test again.

Commit:
bugfix for iovr variables

jkuang-exploro commented:
@JessicaMeixner-NOAA It turns out that's because I included iovr_* variables in the "general" name list(for every CCPP suite). I fixed this issue. Please take another try on your end. I apologize for this mistake.

JessicaMeixner-NOAA commented:
@JianKuang-NOAA the iovr and other namelist options look good.  Thanks for adding the CCPP suite options for  FV3_GFS_v16beta_coupled as well! 

Is the forecast model working again?  Is this ready for me to test again on hera?

jkuang-exploro commented:
@JessicaMeixner-NOAA Yes it is working on my side again.

JessicaMeixner-NOAA commented:
@JianKuang-NOAA my test on orion also succeeded. I started with a fresh clone and build so it's taking a little longer on hera.

@LydiaStefanova-NOAA and @jiandewang could you help check output Lydia note that atm files are now in the "atmos" subfolder
The test on orion output is here: /work/noaa/marine/jmeixner/p6pr194update/test02/COMROOT/test02/gfs.20130401/00 I'll post the hera test as soon as it's available.

JessicaMeixner-NOAA commented:
@JianKuang-NOAA if we wanted to get bit-reproducible answers, we could by overwriting the default values.. however since we're in the process of upgrading, perhaps as long as things are running this is not an issue.  We do want to make sure that everything is working as expected though.

JessicaMeixner-NOAA commented:
@jiandewang @LydiaStefanova-NOAA the hera test output is here: 
/scratch2/NCEPDEV/climate/Jessica.Meixner/p6pr194update/test01/COMROOT/test01/gfs.20130401/00 

The hera test ran to "succeed" although the archive job is still running.   The archive job is not completely correct as @jiandewang pointed out earlier but we can address that in a later PR.

JessicaMeixner-NOAA commented:
Copying @LydiaStefanova-NOAA's comment's to the PR from the issue: 
In atmos/ the grib2 files look ok to me.

In wave/ the gfswave.t00z.global.0p50.fXXX.grib2 files all fields I checked are constant (i.e. identical value at any lat/lon)

Two other things I noticed, that may or may not be relevant here, mentioning them in case they are:
(a) The variable called "la_mod" in the waveocn_*nc files has possibly strange values (e.g the max is 5.37e+34 in wavocn_2013_04_01_03.nc, 8.9301E+08 in wavocn_2013_04_02_21.nc). I haven't looked at this type of files before, so I don't know if this is normal.
(b) the archlist contents (now under atmos/archlist) are different from what we had for P5. I can describe the differences in more detail if this is relevant now.

And from @jiandewang: 
checking ocean outputs show reasonable value (both orion and hera), but confirmed with Lydia's point about LA_MOD

And from me: 
 We know that the arch list is not perfect and will take care of that later. The bad wave grib files is another known issue that this PR is not supposed to address. the la_mod does have very large values at times. That was there before. 

Thanks to @jiandewang and @LydiaStefanova-NOAA for helping check things.

JessicaMeixner-NOAA reviewed (approved):
Thanks @JianKuang-NOAA
